### PR TITLE
Feature/req 50 local admins

### PR DESCRIPTION
**Pull request by viktorianer**
It is possible to create a “Local Administrator” account.

A local administrator can create groups and profiles, manage group privileges (permissions) and access rights for other users in the areas covered by his write rights. The write rights are assigned by other administrators.

Limited allowance of local administrators:

It is not permitted for local administrators to create, edit or deactivate the accounts of other administrators or the Scope Only-Administrators.

Local administrator cannot edit default profiles and groups. The account groups may edit their access rights on their own.  

Local administrators cannot extend their own rights and may assign to other users only the rights that they own themselves.

### CODE REVIEW

File: sernet.gs.server/src/sernet/gs/server/security/LdapAuthenticatorImpl.java [hunk 1]
```diff
@@ -255,6 +255,13 @@ private DirContextOperations ldapUser(Entity entity, String[] specialRoles) {
             }
             roles.add(ApplicationRoles.ROLE_ADMIN);
         }
+        // if set in the entity, the user may also have the local admin role:
+        if (entity.isSelected(Configuration.PROP_ISLOCALADMIN, "configuration_islocaladmin_yes")) {
```
Comment (viktorianer) [inline sernet.gs.server/src/sernet/gs/server/security/LdapAuthenticatorImpl.java]: It is out of scope to refactor LdapAuthenticatorImpl.ldapUser(...) method.

File: sernet.gs.server/src/sernet/verinice/service/ConfigurationService.java [hunk 2]
```diff
@@ -274,19 +285,72 @@ public String getName(String user) {
...
+            LOG.error("Error while checking write permissions", re);
+            throw re;
+        } catch (CommandException t) {
+            LOG.error("Error while checking write permissions", t);
+            throw new RuntimeException("Error while checking write permissions", t);
+        }
+        return false;
+    }
+
+    public Set<String> loadRoles() throws CommandException {
```
Comment (viktorianer) [inline sernet.gs.server/src/sernet/verinice/service/ConfigurationService.java]: And we get circular reference between AccountService and ConfigurationService:

https://docs.spring.io/spring/docs/3.0.x/javadoc-api/org/springframework/beans/factory/FactoryBeanNotInitializedException.html

File: sernet.gs.service/src/sernet/verinice/service/commands/UpdatePermissions.java [hunk 3]
```diff
@@ -97,11 +99,14 @@ private void addParameter(Set<Permission> permissionAdd, Set<Permission> permiss
         this.stationId = ChangeLogEntry.STATION_ID;
     }
 
+    @Override
     public void execute() {
         CnATreeElement cte = loadElement();
-        updateElement(cte);
-        if (updateChildren) {
-            updateChildren(cte.getChildren());
+        if (getConfigurationService().isWriteAllowed(cte)) {
```
Comment (viktorianer) [inline sernet.gs.service/src/sernet/verinice/service/commands/UpdatePermissions.java]: It is out of scope to refactor IRightsService and RightsServiceClient.isEnabled

File: sernet.gs.server/src/sernet/verinice/service/ConfigurationService.java (hunk 2 above)
Comment (pixeldrama) [inline sernet.gs.server/src/sernet/verinice/service/ConfigurationService.java]: Ok, I'll think this goes to deep and won't be fixed within this pull request.

File: sernet.gs.service/src/sernet/verinice/service/commands/UpdatePermissions.java (hunk 3 above)
Comment (pixeldrama) [inline sernet.gs.service/src/sernet/verinice/service/commands/UpdatePermissions.java]: Alright.

**Pull request merged**